File: CitizenScienceBackend/src/main/java/com/citizenScience/dto/UpdateNotesRequest.java
package com.citizenScience.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Data Transfer Object for updating sighting notes.
 * Contains the new note text.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpdateNotesRequest {
    private String note;
}

File: citizen_science/lib/dto/update_notes_request.dart
/// Data transfer object for updating sighting notes.
///
/// Contains the updated note text for a sighting.
class UpdateNotesRequest {
  final String note;

  UpdateNotesRequest({required this.note});

  /// Converts this request to a JSON map for API transmission.
  Map<String, dynamic> toJson() {
    return {'note': note};
  }
}
